core_backend/app/question_answer/routers.py

```diff
@@ -235,6 +237,17 @@ async def search(
         contents=response.search_results,
         asession=asession,
     )
+    if is_unable_to_generate_response(response):
+        failure_reason = response.debug_info["factual_consistency"]
+        response = await retry_search(
+            query_refined=user_query_refined_template,
+            response=response_template,
+            user_id=user_db.user_id,
+            n_similar=int(N_TOP_CONTENT),
+            asession=asession,
+            exclude_archived=True,
+        )
+        response.debug_info["past_failure"] = failure_reason
```

Comment:
Added that in case it works after the second try to  understand why it failed first.